rust/gitxet/tests/integration_tests/test_integration_scenarios.sh

```diff
@@ -0,0 +1,516 @@
+pushd repo_a_4
+git remote add xet_try ../origin_x1
+git fetch xet_try
+git merge --no-edit xet_try/main
+
+create_data_file d2.dat 1000
+cp d2.dat ../
+git add d2.dat
+git commit -a -m "Added d2.dat"
+# Push up to xet_try
+git push xet_try
+
+git fetch origin
+git merge --no-edit origin/main
+
+git push origin
+popd
+
+# STATE:
+#  origin_a: XET, D1 + D2
+#    repo_a_1: XET, D1
+#    repo_a_2: XET, D1
+#    repo_a_3: XET, D1
+#    repo_a_4: XET, D1 + D2
+#    repo_a_5: NO XET, no data
+#    repo_a_X1: XET, no data
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, D2 
+#  origin_x2: XET, no data 
+
+# Now, make sure that repo_a_1 can get the new data from origin_a. 
+pushd repo_a_1
+git fetch origin
+git merge --no-edit origin/main 
+popd
+
+assert_files_equal repo_a_1/d2.dat d2.dat
+
+# Now, make sure that repo_a_2 can get the new data from origin_x1. 
+pushd repo_a_2
+git remote add xet_try ../origin_x1
+git fetch xet_try
+git merge --no-edit xet_try/main
+popd
+
+assert_files_equal repo_a_2/d2.dat d2.dat
+assert_files_equal repo_a_1/d1.dat d1.dat
+
+# STATE:
+#  origin_a: XET, D1 + D2
+#    repo_a_1: XET, D1 + D2
+#    repo_a_2: XET, D1 + D2
+#    repo_a_3: XET, D1
+#    repo_a_4: XET, D1 + D2
+#    repo_a_5: NO XET, no data
+#    repo_a_X1: XET, no data
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, D2 
+#  origin_x2: XET, no data 
+
+
+# Create a new repo based off of x1 
+git clone origin_x1 repo_x1_1  # Will have D2 only
+git clone origin_x1 repo_x1_2  # Will have D2 only
+assert_files_equal repo_x1_1/d2.dat d2.dat 
+
+# Do a a pull from upstream 
+pushd repo_x1_1
+git remote add upstream ../origin_a
+git fetch upstream
+git merge --no-edit upstream/main
+git push origin
+
+# Add new data here.
+create_data_file d3.dat 1000
+cp d3.dat ../d3.dat
+git add d3.dat
+git commit -a -m "Added d3.dat"
+git push origin
+popd
+
+# STATE:
+#  origin_a: XET, D1 + D2
+#    repo_a_1: XET, D1 + D2
+#    repo_a_2: XET, D1 + D2
+#    repo_a_3: XET, D1
+#    repo_a_4: XET, D1 + D2
+#    repo_a_5: NO XET, no data
+#    repo_a_X1: XET, no data
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, D2 + D1 + D3 
+#    repo_x1_1: XET, D1 + D2 + D3
+#    repo_x1_2: XET, D2
+#  origin_x2: XET, no data 
+
+# Now, simulate a pr merge!   
+simulate_pr_merge origin_x1 origin_a
+
+# See if that works with a clone
+git clone origin_a repo_a_6
+assert_files_equal repo_a_6/d1.dat d1.dat 
+assert_files_equal repo_a_6/d2.dat d2.dat 
+assert_files_equal repo_a_6/d3.dat d3.dat 
+
+# See if that works with a merge 
+pushd repo_a_1
+git fetch origin
+git merge --no-edit origin/main 
+popd
+assert_files_equal repo_a_1/d1.dat d1.dat 
+assert_files_equal repo_a_1/d2.dat d2.dat 
+assert_files_equal repo_a_1/d3.dat d3.dat 
+
+# See if it works with a merge from something older. 
```

Comment:
According to the state repo_a_1 and repo_a_2 should be at the same state before fetch&merge?